API/application/iotjs.py

```diff
@@ -123,39 +111,63 @@ def build(self, device):
             build_flags.append('--target-board=%s' % device.get_type())
             build_flags.append('--jerry-heaplimit=78')
-            build_flags.append('--jerry-memstat')
             build_flags.append('--no-parallel-build')
             build_flags.append('--nuttx-home=%s' % paths.NUTTX_PATH)
             build_flags.append('--profile=%s' % utils.join(paths.IOTJS_TEST_PROFILES_PATH,
                                                            'nuttx.profile'))
 
             enable_memstat = True
+            mapfile = utils.join(paths.NUTTX_PATH, "arch/arm/src/nuttx.map")
 
         elif device.get_type() == 'rpi2':
             build_flags.append('--target-board=%s' % device.get_type())
-            build_flags.append('--jerry-cmake-param=-DFEATURE_VALGRIND_FREYA=ON')
-            build_flags.append('--compile-flag=-g')
-            build_flags.append('--jerry-compile-flag=-g')
             build_flags.append('--profile=%s' % utils.join(paths.IOTJS_TEST_PROFILES_PATH,
                                                            'rpi2-linux.profile'))
+            mapfile = utils.join(paths.IOTJS_BUILD_PATH % self.buildtype, "../iotjs.map")
 
         elif device.get_type() == 'artik053':
+            build_flags.append('--profile=%s' % utils.join(paths.IOTJS_TEST_PROFILES_PATH,
+                                                           'tizenrt.profile'))
+
             enable_memstat = True           
+            mapfile = utils.join(paths.TIZENRT_BUILD_PATH, "output/bin/tinyara.map")
 
         else:
             console.fail('Non-minimal IoT.js build failed, unsupported '
                          'device (%s)!' % device.get_type())
 
+        # 1. Run the buildscript and measure binary size
+        # For artik053, build IoT.js in os.build.
+        if not device.get_type() == 'artik053':
+            utils.execute(paths.IOTJS_PATH, 'tools/build.py', build_flags)
+
+        os.build(self, self.buildtype, build_flags, 'all')
+
+        utils.mkdir(paths.IOTJS_MAP_DIR_PATH)
+        utils.copy_file(mapfile, paths.IOTJS_MAP_FILE_PATH)
+
+        os.prebuild(self)
         # Enable memstat for IoT.js (libtuv, jerryscript, iotjs) 
         if enable_memstat:
+            build_flags.append('--jerry-memstat')
             self.__apply_patches()
+        else:
+            build_flags.append('--jerry-cmake-param=-DFEATURE_VALGRIND_FREYA=ON')
+            build_flags.append('--compile-flag=-g')
+            build_flags.append('--jerry-compile-flag=-g')
 
-        # Run the buildscript.
+        # 2. Run the buildscript with memstat or Freya
         # For artik053, build IoT.js in os.build.
         if not device.get_type() == 'artik053':
             utils.execute(paths.IOTJS_PATH, 'tools/build.py', build_flags)
 
-        os.build(self, self.buildtype, 'all')
+        os.build(self, self.buildtype, build_flags, 'all')
+
+        # Revert all the memstat patches from the project.
+        if enable_memstat:
```

Comment:
This block seems duplicated to [line 173](https://github.com/Samsung/js-remote-test/pull/53/files#diff-43d4ce9fbbfdfaf06d8aeb27c6902667R173).